drivers/watchdog/wdt_andes_atcwdt200.c

```diff
@@ -0,0 +1,371 @@
+/*
+ * Copyright (c) 2023 Andes Technology Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT andestech_atcwdt200
+
+#include <zephyr/kernel.h>
+#include <soc.h>
+#include <zephyr/drivers/watchdog.h>
+
+#define LOG_LEVEL CONFIG_WDT_LOG_LEVEL
+#include <zephyr/logging/log.h>
+#include <zephyr/irq.h>
+#include <zephyr/drivers/counter.h>
+#include <zephyr/drivers/syscon.h>
+LOG_MODULE_REGISTER(wdt_andes);
+
+/* Watchdog register */
+#define REG_IDR			0x00
+#define REG_CTRL		0x10
+#define REG_RESTAR		0x14
+#define REG_WREN		0x18
+#define REG_STATUS		0x1c
+
+#define WDT_BASE		(((const struct wdt_atcwdt200_config *)(dev)->config)->base)
```

Comment:
`WDT_BASE(dev)` maybe?

Reply:
we actually do not allow for such helper macros (they were removed treewide from many drivers a long time ago)

Reply:
> `WDT_BASE(dev)` maybe?

OK, I have update the source code

Thanks.

Reply:
> we actually do not allow for such helper macros (they were removed treewide from many drivers a long time ago)

@carlocaione
Thanks for your review and approved.
But after your approvement, we still do some modification this request.
Sorry for any inconvenience caused.